db/migrate/20200809160820_create_tickets.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class CreateTickets < ActiveRecord::Migration[6.0]
+  def change
+    create_table :tickets do |t|
+      t.references :contact, null: true, foreign_key: true
+      t.string :ticket_id
+      t.references :participating_seller, references: :sellers, null: false, foreign_key: { to_table: :sellers}
+      t.references :sponsor_seller, references: :sellers, null: true, foreign_key: { to_table: :sellers}
```

Comment:
i think sponsor seller isn't going to be related to sellers, but participating sellers can be but is nullable!

Reply:
participating sellers will reference seller directly, so both should just reference the corresponding participating/sponsor seller. This should have no reference to seller directly

Reply:
also i think this should be null: false since you can only redeem a ticket at a sponsor seller

Reply:
Updated with the right references, apologies for the earlier confusion! @justintmckibben, could you explain your last comment? Isn't it the opposite? Every ticket needs a ParticipatingSeller (ticket is associated with a merchant that gives it out) but starts null with a SponsorSeller (reward)?

Reply:
I think we figured it out :)